kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/interceptors/BeforeSpecListenerInterceptor.kt

```diff
@@ -10,8 +10,12 @@ import io.kotest.engine.spec.SpecExtensions
 import io.kotest.engine.spec.interceptor.ref.BeforeSpecState
 import io.kotest.engine.spec.interceptor.ref.beforeSpecStateKey
 import io.kotest.engine.test.createTestResult
+import kotlinx.coroutines.sync.Semaphore
+import kotlinx.coroutines.sync.withPermit
 import kotlin.time.Duration.Companion.seconds
 
+private val mutex = Semaphore(1)
```

Comment:
Wouldn't a plain `Mutex()` better express intent?

Reply:
I didn't know that existed, so thanks!